packages/gatsby/cache-dir/loader.js

```diff
@@ -165,19 +164,14 @@ const queue = {
   empty: () => {
     resourcesCount = Object.create(null)
     resourcesArray = []
-    pages = []
-    pathPrefix = ``
+    pathPrefix = `/`
   },
 
   addPagesArray: newPages => {
-    pages = newPages
-    if (
-      typeof __PREFIX_PATHS__ !== `undefined` &&
-      typeof __PATH_PREFIX__ !== `undefined`
-    ) {
-      if (__PREFIX_PATHS__ === true) pathPrefix = `${__PATH_PREFIX__}/`
+    if (__PREFIX_PATHS__) {
+      pathPrefix = `${__PATH_PREFIX__}/`
     }
-    findPage = pageFinderFactory(newPages, pathPrefix)
+    findPage = pageFinderFactory(newPages, pathPrefix.slice(0, -1))
```

Comment:
Why not just use `__PATH_PREFIX__` here? The slicing is confusing.

Reply:
because we use `slice` in other places, when calculating then `basename`, I just want to keep consistency, I agree it's a bit redundant, but I think we can do it in another `pathPrefix` refactor which I described in the issues's description

Reply:
ah right :-) You did cover this in the description 👍to that